### PR TITLE
docs: Corrected typos and grammatical errors

### PR DESCRIPTION
**Pull request by Neilblaze**
### Description

A few typos and grammatical errors were discovered and have been corrected.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

cc: @navendu-pottekkat, @juzhiyuan, @SylviaBABY

### CODE REVIEW

File: docs/en/latest/discovery/nacos.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ $ curl http://127.0.0.1:9180/apisix/admin/stream_routes/1 -H 'X-API-KEY: edd1c9f
 
 #### Specify the namespace
 
-Example of routing a request with an URI of "/nacosWithNamespaceId/*" to a service which name, namespaceId "http://192.168.33.1:8848/nacos/v1/ns/instance/list?serviceName=APISIX-NACOS&namespaceId=test_ns" and use nacos discovery client in the registry:
+Example of routing a request with an URI of "/nacosWithNamespaceId/*" to a service whose name, namespaceId "http://192.168.33.1:8848/nacos/v1/ns/instance/list?serviceName=APISIX-NACOS&namespaceId=test_ns" and use nacos discovery client in the registry:
```
Comment (navendu-pottekkat) [inline docs/en/latest/discovery/nacos.md:127]: "with" might be better here and for other places with similar changes.

File: docs/en/latest/plugins/aws-lambda.md [hunk 2]
```diff
@@ -148,7 +148,7 @@ This approach assumes that you have already an IAM user with programmatic access
 
 ### Configuring path forwarding
 
-The `aws-lambda` Plugins also supports URL path forwarding while proxying requests to the AWS upstream. Extensions to the base request path gets appended to the `function_uri` specified in the Plugin configuration.
+The `aws-lambda` Plugins also support URL path forwarding while proxying requests to the AWS upstream. Extensions to the base request path gets appended to the `function_uri` specified in the Plugin configuration.
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/aws-lambda.md:151]:
```suggestion
The `aws-lambda` Plugin also supports URL path forwarding while proxying requests to the AWS upstream. Extensions to the base request path gets appended to the `function_uri` specified in the Plugin configuration.
```

File: docs/en/latest/discovery/consul.md [hunk 3]
```diff
@@ -143,7 +143,7 @@ curl -X PUT 'http://127.0.0.1:8500/v1/agent/service/register' \
 }'
 ```
 
-In some case, same service name exist in different consul servers.
+In some case, same service name might exist in different consul servers.
```
Comment (abs007) [inline docs/en/latest/discovery/consul.md:146]: `In some cases, the same ..`

File: docs/en/latest/discovery/nacos.md (hunk 1 above)
Comment (navendu-pottekkat) [reply docs/en/latest/discovery/nacos.md]: @Neilblaze Could you make this change throughout? I still see some uses of "whose".

Comment (Neilblaze) [reply docs/en/latest/discovery/nacos.md]: ah I see, okay I'll do it.

**Review by SylviaBABY — APPROVED** [review]

**Review by navendu-pottekkat — APPROVED** [review]

**Pull request merged**